build.sbt

```diff
@@ -270,6 +271,15 @@ lazy val server = (project in file("server"))
       "com.h2database" %  "h2" % "2.2.224",
 
       "org.hibernate.orm" % "hibernate-core" % "6.5.0.Final",
+      // For loading properties from the environment
+      "uk.org.webcompere" % "lightweight-config" % "1.3.0",
+
+      // TODO: Determine if project will support MySQL jar; linter currently states
+      // license is imcompatible
+      // "com.mysql" % "mysql-connector-j" % "8.4.0",
+
+      // For PostgreSQL backend
+      "org.postgresql" % "postgresql" % "42.6.0",
```

Comment:
I don't think we want this to be included by default. It's one of the many hibernate-supported databases and no need to make this any better than others (but including it in the build). UC does not need it, correct? If so, whoever wants PostgreSQL, they'll download the jars and make necessary changes. WDYT?

Reply:
I think there's arguments for both.

Considering that a real database is required to run UC in production, I think it might be good practice to include the supported backends officially, by adding them as build dependencies. It would massively simplify using their respective JDBC jars, rather than trying to mount them to the container AND update the classpath.

We could also then _test_ the backends, using docker-compose with each database. I think this relates to my earlier comments about what databases are considering "officially supported", if any.

The alternative of not including the jars still requires the project to implicitly support different databases, but with less control over testing them. Which in turn, could lead to more questions from the community, because to "official" pattern to self-hosting UC in production is less defined.

For example, not including the jars would require us to modify the image's entrypoint to handle arbitrary ENV vars for classpath modifications and require the user to correctly mount the JDBC jar. 

In my opinion, there's no need to support and document that additional interface. It would be quite brittle, and basically untested. Which means it is not as reliable for self-hosting in production, compared to using a "officially supported" and included jar.

To summarize, both approaches have merit, but I think we should hear from other maintainers if the database interface should have "officially supported" connectors.

Reply:
@tdas Can we get some guidance from yourself or another maintainer on if UC should offer support for "official" backends/databases and include those JDBC connector jars in the forthcoming official image? 

I think I see two paths here. Open to any thoughts on other directions as well.

**1. UC does not include "officially supported" databases to persist its application data. Users of the official image would be responsible for the following:**
a. Mount the correct JDBC jars to the container
b. Modify the relevant `classpath` to reference those jars
c. Mount their own `./etc` config files for hibernate to the container

**Pros**: Smaller image, less project dependencies.
**Cons**: Allows ambiguity about the "best" approach to self-hosting with an external database in production. No testing of different connections and a brittle, implicit interface, which means we could break things that users rely on. For example, reorganizing the config file directories would break a users mount in their deployment scripts.

**2. UC includes "officially supported" JDBC jars (eg. MySQL & Postgres) in the official image. Users of the image would then:**
a. configure their connection by passing env vars to the container (possible vars can be added to `.properties` config files with defaults and interpolated in the app OR
b. Mount their own `./etc` config files for hibernate to the container

**Pros**: We could test the different officially supported databases in CI using docker compose. 
**Cons**: Additional jars means a slightly bigger image size for folks who would only use one of the supported databases.